src/main/java/seedu/duke/parser/CommandMetadata.java

```diff
@@ -47,6 +49,20 @@ protected CommandMetadata(String keyword, String[] groupArguments) throws Illega
         this.pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
     }
 
+    protected CommandMetadata(String keyword, String[] groupArguments, String[] groupArgumentFlags)
+            throws IllegalArgumentException {
+        if (keyword == null || groupArguments == null) {
+            throw new IllegalArgumentException("Keyword, regex, and group arguments cannot be null");
+        }
+        this.keyword = keyword;
+        this.groupArguments = groupArguments;
+        this.groupArgumentFlags = groupArgumentFlags;
+
+        this.regex = generateRegex(keyword, groupArguments, groupArgumentFlags);
+        this.regexLength = groupArguments.length + 1; // Keyword + number of Arguments
+        this.pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+    }
+
```

Comment:
Can you help me refactor, by replacing my original constructor code so that the constructor method will not repeat. Please read through the code diff. This works on my end, but please test this code as well!

From `lines 40-72` of your committed code
```
//@@author sxfoo
protected CommandMetadata(String keyword, String[] groupArguments) throws IllegalArgumentException {
    this(keyword, groupArguments, null);
}

//@@author jensonjenkins
protected CommandMetadata(String keyword, String[] groupArguments, String[] groupArgumentFlags)
        throws IllegalArgumentException {
    if (keyword == null || groupArguments == null) {
        throw new IllegalArgumentException("Keyword, regex, and group arguments cannot be null");
    }
    this.keyword = keyword;
    this.groupArguments = groupArguments;

    if (groupArgumentFlags == null) {
        groupArgumentFlags = new String[groupArguments.length];
        Arrays.fill(groupArgumentFlags, "mandatory"); // default flag
    }
    if (groupArgumentFlags.length != groupArguments.length) {
        throw new IllegalArgumentException("Group Argument and Group Argument Flags does not have the same size!");
    }

    this.groupArgumentFlags = groupArgumentFlags;

    this.regex = generateRegex(keyword, groupArguments, groupArgumentFlags);
    this.regexLength = groupArguments.length + 1; // Keyword + number of Arguments
    this.pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
}

//@@author sxfoo
protected String getKeyword() {
    return keyword;
}

Rest of code here...
```

From `lines 90-126` of your commit code
```
DELETE this COMMENTED CODE: 
  /*private static String generateRegex(String keyword, String[] groupArguments) {
        if (keyword == null || groupArguments == null) {
            throw new IllegalArgumentException("Keyword and groupArguments must not be null");
        }

        StringBuilder regexPattern = new StringBuilder(keyword);
        for (String groupArgName : groupArguments) {
            appendRegex(regexPattern, groupArgName);
        }
        return regexPattern.toString();
    } */

    //@@author jensonjenkins
    private static String generateRegex(String keyword, String[] groupArguments, String[] groupArgumentFlags) {
        if (keyword == null || groupArguments == null || groupArgumentFlags == null) {
            throw new IllegalArgumentException("Keyword, groupArguments and flags must not be null");
        }
        assert groupArgumentFlags.length == groupArguments.length : "Arguments and Flags must be of the same size!";

        StringBuilder regexPattern = new StringBuilder(keyword);
        for (int i = 0; i < groupArguments.length; i++) {
            if (groupArgumentFlags[i].equals("optional")) {
                appendOptionalRegex(regexPattern, groupArguments[i]);
            } else {
                appendRegex(regexPattern, groupArguments[i]);
            }
        }
        return regexPattern.toString();
    }

    //@@author sxfoo
    // Helper function for generateRegex
    private static void appendRegex(StringBuilder regexPattern, String groupArgName) {
        String argRegex = argRegexMap.get(groupArgName);
        if (argRegex == null) {
            throw new IllegalArgumentException("No regex pattern found for argument: " + groupArgName);
        }
        regexPattern.append("\\s+").append(argRegex);
    }

    //@@author jensonjenkins
    private static void appendOptionalRegex(StringBuilder regexPattern, String groupArgName) {
        String argRegex = argRegexMap.get(groupArgName);
        if (argRegex == null) {
            throw new IllegalArgumentException("No regex pattern found for argument: " + groupArgName);
        }
        String optionalGroup = "(\\s+" + argRegex + ")?";
        regexPattern.append(optionalGroup);
    }